README.md

```diff
@@ -4,53 +4,47 @@ Deploy a pwn.college dojo instance!
+The pwn.hust.college infrastructure allows users the ability to "start" challenges, which spins up a private docker container for that user.
 This docker container will have the associated challenge binary injected into the container as root-suid, as well as the flag to be submitted as readable only by the the root user.
 Users may enter this container via `ssh`, by supplying a public ssh key in their profile settings, or via vscode in the browser ([code-server](https://github.com/cdr/code-server)).
 The associated challenge binary may be either global, which means all users will get the same binary, or instanced, which means that different users will receive different variants of the same challenge.
 
-## Dependencies
-
-The only dependency to run the infrastructure is docker, which can be installed with:
-
-```sh
-curl -fsSL https://get.docker.com | /bin/sh
-```
-
 ## Setup
 
-First, clone the repository:
-
 ```sh
-git clone https://github.com/HUSTSeclab/dojo/ /opt/dojo
-```
-
-Now, build the container:
-
-```sh
-docker build -t pwncollege/dojo .
-```
-
-Finally, run the infrastructure which will be hosted on domain `my.domain.college` with:
+curl -fsSL https://get.docker.com | /bin/sh
 
-```sh
-docker run --privileged -d -v /opt/dojo:/opt/pwn.college:shared -p 22222:22 -p 8080:80 -p 10443:443 --name dojo pwncollege/dojo
+DOJO_PATH="./dojo"
+git clone https://github.com/HUSTSeclab/dojo.git "$DOJO_PATH"
+docker build -t pwncollege/dojo "$DOJO_PATH"
+docker run --privileged -d -v "${DOJO_PATH}:/opt/pwn.college:shared" -p 22:22 -p 80:80 -p 443:443 --name dojo pwncollege/dojo
 ```
 
-> [!NOTE]
-> This command would map ports(22, 80, 443) in the container to the corresponding ports(22222, 8080, 10443) on the Docker host.
-> If these ports are bound, especially Port 22, you can disable these processes or modify the mapping ports.
+**You can setup dojo using [setup.sh](https://github.com/HUSTSeclab/dojo/blob/hustsec_dev/setup.sh)**
+
+> **Warning**
+> **(MacOS)**
+>
+> It's important to note that while the dojo is capable of operating on MacOS (either x86 or ARM), MacOS has inherent limitations when it comes to nested Linux mounts within a MacOS bind mount. 
+> This limitation specifically affects `data/docker`, which necessitates the use of OverlayFS mounts, preventing nested docker orchestration from functioning properly.
+> In order to circumvent this issue, you must ensure that`data/docker` is not backed by a MacOS bind mount. 
+> This can be accomplished by replacing the bind mount with a docker volume for `data/docker`, which will use a native Linux mount. 
+> You can apply this solution using the following Docker command (notice the additional `-v`):
+>
+> ```sh
+> docker run --privileged -d -v "${DOJO_PATH}:/opt/pwn.college:shared" -v dojo-data-docker:/opt/pwn.college/data/docker -p 22:22 -p 80:80 -p 443:443 --name dojo pwncollege/dojo
+> ```
```

Comment:
We don't need MacOS setting, so please revert this part. Add back our notice of port number.